speed_up_in_r_programming.Rmd

```diff
@@ -0,0 +1,381 @@
+
+$$ 
+  MSE = \frac{\sum_{r=1}^{\text{rep}} (\widehat \mu_r - \mu_{\text{true}})^2}{\text{rep}}
+$$
+
+```{r}
+## check if a given integer is prime
+isPrime = function(n) {
+  if (n <= 3) {
+    return (TRUE)
+  }
+  if (any((n %% 2:floor(sqrt(n))) == 0)) {
+    return (FALSE)
+  }
+  return (TRUE)
+}
+
+## estimate mean only using observation with prime indices
+estMeanPrimes = function(x) {
+  n <- length(x)
+  ind <- sapply(1:n, isPrime)
+  return (mean(x[ind]))
+}
+
+## compare methods: sample avg and prime-indexed avg
+compare_methods <- function(dist = "gaussian", n = 100, reps = 100, seed = 123) {
+  # set seed according to command argument `seed`
+  set.seed(seed)
+  
+  # preallocate space to store estimators
+  msePrimeAvg <- 0.0
+  mseSamplAvg <- 0.0
+  # loop over simulation replicates
+  for (r in 1:reps) {
+    # simulate data according to command arguments `n` and `distr`
+    if (dist == "gaussian") {
+      x = rnorm(n)
+    } else if (dist == "t1") {
+      x = rcauchy(n)
+    } else if (dist == "t5") {
+      x = rt(n, 5)
+    } else {
+      stop(paste("unrecognized dist: ", dist))
+    }
+    # prime indexed mean estimator and classical sample average estimator
+    msePrimeAvg <- msePrimeAvg + estMeanPrimes(x)^2 
+    mseSamplAvg <- mseSamplAvg + mean(x)^2
+  }
+  mseSamplAvg <- mseSamplAvg / reps
+  msePrimeAvg <- msePrimeAvg / reps
+  return(c(mseSamplAvg, msePrimeAvg))
+}
+```
+
+In order to indicate the *option 2*, we save the above codes in script [`runSim.R`]. 
+    
+#### Option 2
+*this example is only good for linux system*
+
+1. The [`runSim.R`] script to include arguments `seed` (random seed), `n` (sample size), `dist` (distribution) and `rep` (number of simulation replicates). When `dist="gaussian"`, generate data from standard normal; when `dist="t1"`, generate data from t-distribution with degree of freedom 1 (same as Cauchy distribution); when `dist="t5"`, generate data from t-distribution with degree of freedom 5. Calling `runSim.R` will (1) set random seed according to argument `seed`, (2) generate data according to argument `dist`, (3) compute the primed-indexed average estimator and the classical sample average estimator for each simulation replicate, (4) report the average mean squared error (MSE)
+
+2. The [`autoSim.R`] script to run simulations with combinations of sample sizes `nVals = seq(100, 500, by=100)` and distributions `distTypes = c("gaussian", "t1", "t5")`.  `rep = 50` is used for MSE. 
+
+```{bash}
+cat resources/spped_up_in_r_programming/autoSim.R
+```
+
+
+3. Download the [`runSim.R`] and [`autoSim.R`] scripts, and run the following codes, this will parallelly compute results from all combinations, and write output to appropriately named files.
+ 
+```{bash}
+    Rscript resources/spped_up_in_r_programming/autoSim.R seed=280 rep=50

```

Comment:
typo in 'spped

Reply:
Thank you! Do I need to resubmit?

Reply:
Thank you! Is it fine now?

Reply:
Hi! We need to confirm with Prof Robbins that the branch name is acceptable. In our instructions the branch name should not be capitalized.